test/integration/cilium-nodesubnet/ipconfigupdate.go

```diff
@@ -100,12 +100,13 @@ func main() {
 		}
 
 		if primaryIPConfig != nil {
-			for i := 2; i <= secondaryConfigCount+1; i++ {
+			ipConfigurations = []interface{}{}
+			for i := 1; i <= secondaryConfigCount+1; i++ {
 				ipConfig := make(map[string]interface{})
 				for k, v := range primaryIPConfig {
 					// only the primary config needs loadBalancerBackendAddressPools. Azure doesn't allow
-					// secondary IP configs to be associated load balancer backend pools.
-					if k == "loadBalancerBackendAddressPools" {
+					// secondary IP configs to be associated load balancer backend pools. Also skip nil values.
+					if k == "loadBalancerBackendAddressPools" || v == nil {
```

Comment:
why are we hardcoding `loadBalancerBackendAddressPools`

Reply:
We do have a comment above, we are copying the primary ip config to create secondary ones, and we need to remove the field. I could define it as a variable, but this is anyway a utility script. Are you suggesting to do that?

Reply:
Resolving this conversation since the hard coding isn't new, can clean up in a follow up if needed.